### ISSUE TITLE
[iOS] App does not start when switch off "Background App Refresh" and "Notifications"

### ISSUE THREAD

**Issue by rubeonline** [phonegap/phonegap-plugin-push#2387]
### Expected Behaviour
When i turn off Background App Refresh and Notifications in my iOS App Settings i don´t want to register for Push Notifications.

### Actual Behaviour
It tries to register for push and my app turns into an endless loop in the splash screen. In the Source files it only arrives to this log "Using APNS Notification". After that nothing happens.

#### Steps to Reproduce 

- Install App
- Turn Off "Background App Refresh" and "Notifications"
- Open App

#### Platform and Version (eg. Android 5.0 or iOS 9.2.1)
iOS 11.3.1

#### Cordova CLI version and cordova platform version

    cordova --version                                    # e.g. 7.1.0
    cordova platform version ios                     # e.g. 4.5.4

#### Plugin version

    cordova plugin version | grep phonegap-plugin-push   # e.g. 2.1.2

#### Sample Code that illustrates the problem

```
 registerForPushNotifications: function () {
            console.log("PushInit Start");
            push = PushNotification.init({
                android: { senderID: '', forceShow : true },
                ios: { alert: 'true', badge: 'false', sound: 'true' },
                wns: {}
            });
            console.log("PushInit End");

            push.on('registration', (data) => {

                console.log("PushRegistration Start");
    
                    var platform = device.platform;
    
                    var notificationPlatform;
                    if (platform == 'android' || platform == 'Android') {
                        notificationPlatform = 'gcm';
                    } else if (device.platform === 'iOS') {
                        notificationPlatform = 'apns';
                    }
                    var getParameters = "someparameters";
                    app.start("", getParameters);
                });
    
            push.on('error', (e) => {
                console.log(e.message);
                alert("Registration for push not possible.")
                app.start("","");
            });


        push.on('notification', (data) => {
            // do sth
        });
    }
```

#### Logs taken while reproducing problem
2018-05-24 15:00:14.720371+0200 TestApp[1248:372670] Push Plugin register called
2018-05-24 15:00:14.720412+0200 TestApp[1248:372670] PushPlugin.register: setting badge to true
2018-05-24 15:00:14.720421+0200 TestApp[1248:372670] PushPlugin.register: clear badge is set to 0
2018-05-24 15:00:14.720429+0200 TestApp[1248:372670] PushPlugin.register: better button setup
2018-05-24 15:00:14.720623+0200 TestApp[1248:372670] FCM Sender ID (null)
2018-05-24 15:00:14.720641+0200 TestApp[1248:372670] Using APNS Notification

**Comment by macdonst** [comment]
@rubeonline I'm confused why you would want to do this an also include the push plugin?

**Comment by jcesarmobile** [comment]
I suppose he means disabling that on the app settings once installed, an user can do that, but the app should continue working even if the push are disabled.

**Comment by macdonst** [comment]
@jcesarmobile yeah, I haven't seen that behaviour when you disable that stuff in iOS settings. My mistaken assumption was it was something @rubeonline was turning off in Xcode. I'll wait for clarification.

**Comment by rubeonline** [comment]
Yes when the user disable these settings in iOS.

**Comment by rubeonline** [comment]
@macdonst Any news on that? Can i provide sth more?
